### ISSUE TITLE
Why are the validation sets augmented

### ISSUE THREAD

**Issue by Tinghh** [ChengBinJin/V-GAN-tensorflow#13]
please... when running the program,the training image is 9,and the validation images is 240
I don't undersdand the augmented images are divided into train/validation set,why the validation be augmented...but train set is so small..

**Comment by ChengBinJin** [comment]
training images are used online data augmentation because of the memory issue. For example, each iteration randomly selects one image then randomly flip, translate, and rotate the image. However, for the validation images, each epoch we want to do validation to check the mode's performance. And the validation data is much smaller than the training data. So for the validation data, we do off-line data augmentation and save them on the memory. The benefit of the off-line data augmentation is saving time due to the one time process. Therefore, '9' in your question is the original training images before on-line data augmentation, '240' is the number of validation data that is augmented.

**Comment by Tinghh** [comment]
Thank you for your reply!
So for DRIVE， the training images are augmented to 50020 after 50000 iterations? 
Another question,why is the same image  in the  files of sample ? Isn't it random?
And what's the meaning of  the number in pictures'name,For example, 0_55, 200_227, 400_121,   the number of  55, 227,121, What does them stand for?

**Comment by ChengBinJin** [comment]
The training images are augmented as the number of training iterations using on-line data augmentation. 
Training data is separated into the training and validation data. Only 10 % of the data is selected for validation data because the number of data is too little. It's the reason why the same validation data always sampled. For example, we have 20 training images. Then, 18 images and the 2 images for validation only.
For file name 0_55, 0 means 'number of iteration' and 55 means that ID of the validation image in off-line augmentation. So the format is 'iter-time_val-img-id'


> Thank you for your reply!
> So for DRIVE， the training images are augmented to 50020 after 50000 iterations?
> Another question,why is the same image in the files of sample ? Isn't it random?
> And what's the meaning of the number in pictures'name,For example, 0_55, 200_227, 400_121, the number of 55, 227,121, What does them stand for?

**Comment by Tinghh** [comment]
Thanks again for your reply！
I have a new problem.When I was doing the evalution,the auc_roc and auc_pr of V-gan is very low ,but another evaluation index，such as ,specificity,acc,and sensitivity are high than other model.
I don't know why，，，
![image](https://user-images.githubusercontent.com/50097443/57425270-44925600-724d-11e9-929b-96c6ad4df45e.png)

**Comment by ChengBinJin** [comment]
> Thanks again for your reply！
> I have a new problem.When I was doing the evalution,the auc_roc and auc_pr of V-gan is very low ,but another evaluation index，such as ,specificity,acc,and sensitivity are high than other model.
> I don't know why，，，
> ![image](https://user-images.githubusercontent.com/50097443/57425270-44925600-724d-11e9-929b-96c6ad4df45e.png)

I also found a similar problem. 
The code used roc_curve and roc_auc_score functions from the sklearn.metrics library.
I think the problem with these two functions.
These functions seem to calculate the relative value, not the absolute measurement.
![img](https://user-images.githubusercontent.com/37034031/57566447-a696bb00-7407-11e9-8a3c-d8d05bd39f03.png)

We hope that our model saved according to the best evaluation score. This score is a summation of the roc_curve and roc_auc_score. But the problem is the calculation of the two functions are the same as the right side image (relative). But we need to select the best model according to the left side (absolute) calculation. I just have 70~80 % confidence to make sure the problem.

**Comment by little1TouwaErio** [comment]
Tinghh,I have some questions about my understanding of this code. Could you please tell me your email address?I have some questions to ask you. Looking forward to your reply. Thank you very much! @Tinghh

**Comment by Tinghh** [comment]
1358331736@qq.com @Lililicg

**Comment by Tinghh** [comment]
Although I still have some questions, thank you very much for your reply and patient answer. You are really a handsome man！ @ChengBinJin

**Comment by Lililicg** [comment]
------------------ 原始邮件 ------------------
发件人: "Tinghh"<notifications@github.com>;
发送时间: 2019年5月29日(星期三) 上午8:46
收件人: "ChengBinJin/V-GAN-tensorflow"<V-GAN-tensorflow@noreply.github.com>;
抄送: "416580793"<416580793@qq.com>;"Mention"<mention@noreply.github.com>;
主题: Re: [ChengBinJin/V-GAN-tensorflow] Why are the validation setsaugmented (#13)

请问我现在方便联系你吗

1358331736@qq.com @Lililicg
